convert2rhel/pkgmanager/handlers/yum.py

```diff
@@ -0,0 +1,272 @@
+                _resolve_yum_problematic_dependencies(msg)
+                return False
+        return True
+
+    def _process_transaction(self):
+        """Internal method to process the transaction.
+
+        :raises SystemExit: If we can't process the transaction.
+        """
+        try:
+            self._base.processTransaction()
+        except (
+            pkgmanager.Errors.YumRPMCheckError,
+            pkgmanager.Errors.YumTestTransactionError,
+            pkgmanager.Errors.YumRPMTransError,
+        ) as e:
+            loggerinst.debug("Got the following exception message: %s", e)
+            loggerinst.critical("Failed to validate the yum transaction.")
+
+        loggerinst.info("Transaction processed succesfully.")
+
+    def process_transaction(self, test_transaction=False):
+        """Process the yum transaction.
+
+        This function is supposed to be an replacement of the old
+        `call_yum_cmd_w_downgrades()` function, that in the past, used to call
+        the yum commands (upgrade, reinstall and downgrade) several times in
+        order to replace all the possible packages from the original system
+        vendor to the RHEL ones.
+
+        ..notes::
+            The implementation of this yum transaction is different from the
+            dnf one, mainly because yum can raise some "exceptions" while
+            trying to resolve the dependencies of the transaction. Because of
+            this, we need to loop throught a couple of times until we know that
+            all of the dependencies are resolved without problems.
+
+            You might wonder "why not removing the packages that caused a
+            failure and only loop throught the dep solving again?" Well. Since
+            we are removing the problematic packages using `rpm` and not some
+            specific method in the transaction itself, yum doesn't know that
+            something has changed (The resolveDeps() function doesn't refresh
+            if something else happens outside the transaction), in order to
+            make sure that we won't have any problems with our transaction, it
+            is easier to loop throught everything again and just recreate the
+            transaction, so yum will keep track of what's changed.
+
+            This function should loop max 3 times to get to the point where our
+            transaction doesn't have any problematic packages in there, and the
+            subsequent transactions are "faster" because of some yum internal
+            cache mechanism.
+
+            This might be optimizaed in the future, but for now, it's somewhat
+            reliable.
+
+        :param test_transaction: Determines if the transaction needs to be
+        tested or not.
+        :type test_transaction: bool
+        :raises SystemExit: If we can't resolve the transaction dependencies.
+        :return: A boolean indicating if it was successful or not.
+        :rtype: bool
+        """
```

Comment:
```suggestion
        """Process the yum transaction.

        This function is supposed to be an replacement of the old
        `call_yum_cmd_w_downgrades()` function, that in the past, used to call
        the yum commands (upgrade, reinstall and downgrade) several times in
        order to replace all the possible packages from the original system
        vendor to the RHEL ones.

        ..notes::
            The implementation of this yum transaction is different from the
            dnf one, mainly because yum can raise some "exceptions" while
            trying to resolve the dependencies of the transaction. Because of
            this, we need to loop through a couple of times until we know that
            all of the dependencies are resolved without problems.

            You might wonder "why not remove the packages that caused a
            failure and loop through the dep solving again?" Well. Since
            we are removing the problematic packages using `rpm` and not some
            specific method in the transaction itself, yum doesn't know that
            something has changed (The resolveDeps() function doesn't refresh
            if something else happens outside the transaction), in order to
            make sure that we won't have any problems with our transaction, it
            is easier to loop through everything again and just recreate the
            transaction, so yum will keep track of what's changed.

            This function should loop max 3 times to get to the point where our
            transaction doesn't have any problematic packages in there, and the
            subsequent transactions are "faster" because of some yum internal
            cache mechanism.

            This might be optimized in the future, but for now, it's somewhat
            reliable.

        :param test_transaction: Determines if the transaction needs to be
        tested or not.
        :type test_transaction: bool
        :raises SystemExit: If we can't resolve the transaction dependencies.
        :return: A boolean indicating if it was successful or not.
        :rtype: bool
        """
```